include/utility.hpp

```diff
@@ -272,11 +264,8 @@ namespace giml {
             return *this;
         }
 
-        ~CircularBuffer() {
-            if (this->pBackingArr) {
-                free(pBackingArr);
-            }
-        }
+        // Destructor that frees the memory
+        ~CircularBuffer() {if (this->pBackingArr) {free(pBackingArr);}}
```

Comment:
This honestly reduces readability. If you are trying to reduce code size, a couple lines of whitespace will not have much effect, and all this compiles to the same thing regardless. Readability is more important here.

Reply:
I added the spacing. Look better now? I think the logic on this one is simple enough to be best read as a one-liner.